app/src/main/java/hm/orz/chaos114/android/tumekyouen/modules/create/CreateActivity.kt

```diff
@@ -138,47 +143,41 @@ class CreateActivity : DaggerAppCompatActivity(), CreateKyouenView.CreateKyouenV
             show();
         }
 
-        tumeKyouenService.postStage(data)
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
-            .autoDisposable(scopeProvider)
-            .subscribe(
-                { response ->
-                    val responseString = response.body()?.string()
-                    Timber.d("sucess : %s", responseString)
-                    progressDialog.dismiss()
-
-                    val message = convertResponseToMessage(responseString)
-                    AlertDialog.Builder(this)
-                        .setMessage(message)
-                        .setPositiveButton(android.R.string.ok, null)
-                        .show()
-                },
-                { throwable ->
-                    Timber.d(throwable, "fail")
-                    AlertDialog.Builder(this)
-                        .setMessage(R.string.create_result_failure)
-                        .setPositiveButton(android.R.string.ok, null)
-                        .show()
-                }
-            )
-    }
-
-    private fun convertResponseToMessage(response: String?): String {
-        if (response == null) {
-            return getString(R.string.create_result_failure)
-        }
+        val newStage = NewStage(size.toLong(), stage, creator)
+        GlobalScope.launch(Dispatchers.Main) {
+            val response = withContext(Dispatchers.Default) {
+                tumeKyouenV2Service.postStage(newStage)
+            }
 
-        val matchEntire = "success stageNo=([0-9]*)".toRegex().matchEntire(response)
-        if (matchEntire == null) {
-            if (response.equals("registered")) {
-                return getString(R.string.create_result_registered)
+            progressDialog.dismiss()
+            if (response.isSuccessful) {
+                showSuccessDialog(response)
+            } else {
+                showFailedDialog(response)
             }
-            return getString(R.string.create_result_failure)
         }
+    }
 
-        return matchEntire.destructured.let { (stageNo) ->
-            getString(R.string.create_send_success_message, stageNo)
+    private fun showSuccessDialog(response: Response<Stage>) {
+        Timber.d("sucess : %s", response.body())
+        val stage = response.body()
+        val message = getString(R.string.create_send_success_message, stage?.stageNo.toString())
+        AlertDialog.Builder(this)
+            .setMessage(message)
+            .setPositiveButton(android.R.string.ok, null)
+            .show()
+    }
+
+    private fun showFailedDialog(response: Response<Stage>) {
+        val message: String
+        if (response.code() == 409) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/noboru-i/kyouen-android/blob/a5c24ec48abc4b18a58802172ea3f43c28a46200/app/src/main/java/hm/orz/chaos114/android/tumekyouen/modules/create/CreateActivity.kt#L173"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>